### PR TITLE
Upgrade to cosmos-sdk v0.40.0-rc5

### LINKED ISSUE #344: Upgrade sdk to v0.40.0-rc5
See https://github.com/cosmos/cosmos-sdk/releases/tag/v0.40.0-rc5

### PR DESCRIPTION
**Pull request by alpe**
Resolves #344 

Note: new zap logger looks a bit funny and verbose

### CODE REVIEW

File: x/wasm/client/cli/new_tx.go [hunk 1]
```diff
@@ -18,8 +18,7 @@ func MigrateContractCmd() *cobra.Command {
 		Short: "Migrate a wasm contract to a new code version",
 		Args:  cobra.ExactArgs(3),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clientCtx := client.GetClientContextFromCmd(cmd)
-			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			clientCtx, err := client.GetClientTxContext(cmd)
```
Comment (ethanfrey) [inline x/wasm/client/cli/new_tx.go:21]: This does look like a cleaner API

File: cmd/wasmd/testnet.go [hunk 2]
```diff
@@ -1,397 +0,0 @@
-package main
-
-// DONTCOVER
-
-import (
```
Comment (ethanfrey) [inline cmd/wasmd/testnet.go:5]: Fair enough, we never used it. Good to chop out dead weight.

File: app/sim_test.go [hunk 3]
```diff
@@ -1,338 +0,0 @@
-package app
-
-import (
```
Comment (ethanfrey) [inline app/sim_test.go:3]: Fair enough to remove the sim tests, which we don't use.

We should probably trim down the circle ci file to match (or maybe you did that already... I think there is a lot of commented out sections there)

File: app/integration/test_helpers.go [hunk 4]
```diff
@@ -1,210 +0,0 @@
-package integration
-
-/**
-This file is full of test helper functions, taken from simapp
```
Comment (ethanfrey) [inline app/integration/test_helpers.go:4]: I thought I imported this from some integration tests in `x/wasm`, but I guess we don't need it any more?

**Review by ethanfrey — COMMENTED** [review]
Thanks for starting this and nice to trim out dead code we don't use.

Question about removing `app/integration/test_helpers.go`, but otherwise looking good.

File: app/integration/test_helpers.go (hunk 4 above)
Comment (alpe) [reply app/integration/test_helpers.go:4]: ja, this was used with the IBC coordinator but can be added again when used.

File: app/sim_test.go (hunk 3 above)
Comment (alpe) [reply app/sim_test.go:3]: I did some cleanup in the past already

**Comment by ethanfrey** [comment]
Will re-review now. The CI is failing when grepping out DONTCOVER.

On master there is only one file that has that label (testnets, which you deleted). https://github.com/CosmWasm/wasmd/search?q=DONTCOVER

I think you can just remove that section and hopefully code coverage works again.

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,22 +4,16 @@
...
 - Invalid homeDir for export command [\#334](https://github.com/CosmWasm/wasmd/issues/334)
 
-**Closed issues:**
-
-- Add Dependabot to keep dependencies secure and up-to-date [\#336](https://github.com/CosmWasm/wasmd/issues/336)
-
-**Merged pull requests:**
-
-- Upgrade grpc lib to v1.34.0 [\#340](https://github.com/CosmWasm/wasmd/pull/340) ([alpe](https://github.com/alpe))
-- Fix param usage issue in genesis import [\#338](https://github.com/CosmWasm/wasmd/pull/338) ([alpe](https://github.com/alpe))
```
Comment (ethanfrey) [inline CHANGELOG.md:20]: The last two here deserve changelog entries, no?

File: x/wasm/client/cli/query.go [hunk 6]
```diff
@@ -65,7 +67,7 @@ func GetCmdListCode() *cobra.Command {
 			if err != nil {
 				return err
 			}
-			return clientCtx.PrintProto(res)
+			return clientCtx.WithJSONMarshaler(&VanillaStdJsonMarshaller{}).PrintProto(res)
```
Comment (ethanfrey) [inline x/wasm/client/cli/query.go:70]: Mmmmm... I like vanilla

File: x/wasm/client/cli/query.go [hunk 7]
```diff
@@ -349,14 +351,13 @@ func GetCmdGetContractHistory() *cobra.Command {
...
 		RunE: func(cmd *cobra.Command, args []string) error {
-
 			clientCtx, err := client.GetClientQueryContext(cmd)
 			if err != nil {
 				return err
 			}
 
 			_, err = sdk.AccAddressFromBech32(args[0])
-			if err == nil {
+			if err != nil {
```
Comment (ethanfrey) [inline x/wasm/client/cli/query.go:360]: Yeah, totally important typo

File: x/wasm/client/cli/query.go [hunk 8]
```diff
@@ -428,6 +429,41 @@ func asciiDecodeString(s string) ([]byte, error) {
...
+func (x VanillaStdJsonMarshaller) MarshalInterfaceJSON(i proto.Message) ([]byte, error) {
+	any, err := codectypes.NewAnyWithValue(i)
+	if err != nil {
+		return nil, err
+	}
+	return x.MarshalJSON(any)
+}
+
+func (x VanillaStdJsonMarshaller) MarshalJSON(o proto.Message) ([]byte, error) {
+	return json.MarshalIndent(o, "", " ")
```
Comment (ethanfrey) [inline x/wasm/client/cli/query.go:444]: I like this, indented by default

**Review by ethanfrey — APPROVED** [review]
Looks good. Just the CI issue I mention above.

File: CHANGELOG.md (hunk 5 above)
Comment (alpe) [reply CHANGELOG.md:20]: The pull requests are actually duplicates of the issues. That's why I removed them here.

**Comment by alpe** [comment]
> Will re-review now. The CI is failing when grepping out DONTCOVER.

I restarted the failing step and it succeeded. From the circleCI output it was not clear to me if the grep failed or the upload.

**Comment by ethanfrey** [comment]
Great. Looking good.

Seeing the x here, wondering about disabling (or modifying) the code coverage checks.

You increase coverage in this PR, but add some misses and it marks it as a fail?

**Comment by alpe** [comment]
ja, the overall coverage increased but the coverage of the diff in this PR was not high enough. 
https://docs.codecov.io/docs/codecov-delta#examples
I have an 👁️  on this for feature PRs but with this upgrade I did not add more tests than we had.

**Pull request merged**